### PR TITLE
Add recommended Elixir x Erlang/OTP combinations to CI

### PR DESCRIPTION
**Pull request by adrianomitre**
Quoting @josevalim:
> I would still advise folks to go with:
> * For each Elixir version, add a run with earliest supported Erlang/OTP
> * For the last Elixir version, also test the latest supported Erlang/OTP

Please refer to the original discussion on https://github.com/dashbitco/broadway/pull/188 for the full rationale.

Please refer to [Yet another Elixir and Erlang docker image - Chat / Discussions - Elixir Forum](https://elixirforum.com/t/yet-another-elixir-and-erlang-docker-image/28740) for the rationale behing Elixir images by [Hexpm's Bob](https://github.com/hexpm/bob#docker-images).

### CODE REVIEW

**Comment by josevalim** [comment]
@adrianomitre the MySQL 8.0 dump tests are failing. Can you please take a look?

**Comment by josevalim** [comment]
@adrianomitre unfortunately the mysqldump tests are still failing. Any idea why?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -47,7 +64,8 @@ jobs:
           apt-get update
           apt-get install --allow-unauthenticated -y mysql-client
```
Comment (Eiji7) [inline .github/workflows/ci.yml:65]: @adrianomitre On CI there is such error:
> ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)

Following [this answer](https://stackoverflow.com/a/15039113) I suggest to change this line:
```suggestion
          apt-get install --allow-unauthenticated -y mysql-server
```

If that fix would work then probably we should change `step` name to `Install MySQL Server`.

**Comment by josevalim** [comment]
Btw, some MySQL failures also happen on master. Only the dump ones are unexpected!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (adrianomitre) [reply .github/workflows/ci.yml:65]: @Eiji7 could you please point out in which commit, job, step such error ocurred?

I could not find it on step `Run MYSQL_URL=root:root@mysql ECTO_ADAPTER=myxql mix test` of the [job `test-mysql (8.0)` of commit 1debbfd9](https://github.com/elixir-ecto/ecto_sql/pull/242/checks?check_run_id=910137830).

Comment (adrianomitre) [reply .github/workflows/ci.yml:65]: @Eiji7 Also, it seems that MySQL server is running on a [service container](https://docs.github.com/en/actions/configuring-and-managing-workflows/about-service-containers), running `mysql:${{ matrix.mysql_version }}` image (see https://github.com/elixir-ecto/ecto_sql/pull/242/files#diff-e9f950f17198d3d5e3122a44230a09b9L29).

**Comment by adrianomitre** [comment]
> Btw, some MySQL failures also happen on master. Only the dump ones are unexpected!

It seems to be non-deterministic, as it has passed on the following workflow: https://github.com/elixir-ecto/ecto_sql/pull/242/checks?check_run_id=910257031

**Comment by josevalim** [comment]
Closing this for now as it has gone stale. Thank you!

**Pull request closed without merging**